ner_v2/detectors/numeral/number_range/standard_number_range_detector.py

```diff
@@ -0,0 +1,385 @@
+            [In]  >> _tag_number_in_text(text)
+            [Out] >> 'i want to buy __number__1 apples and more than __number__2 bananas'
+        """
+        tagged_number_text = processed_text
+        sorted_number_detected_map = sorted(self.number_detected_map.items(), key=lambda kv: len(kv[1][1]),
+                                            reverse=True)
+        for number_tag in sorted_number_detected_map:
+            tagged_number_text = tagged_number_text.replace(number_tag[1][1], number_tag[0], 1)
+        return tagged_number_text
+
+    def _get_number_tag_dict(self):
+        """
+        Method to create number tag dict. Its run number detection on text and create a dict having number tag as key
+        and value as tuple of entity value and original text.
+        Returns:
+            detected_number_dict(dict): dict containing number tag and their corresponding value and original text
+        Examples:
+            [In]  >> text = 'I want 12 dozen banana'
+            [In]  >> _get_number_tag_dict()
+            [Out] >> {'__number_1': ({'value': 12, 'unit': None}, '12')}
+        """
+        detected_number_dict = {}
+        entity_value_list, original_text_list = self.number_detector_object.detect_entity(self.processed_text)
+        index = 1
+        for entity_value, original_text in zip(entity_value_list, original_text_list):
+            detected_number_dict[NUMBER_REPLACE_TEXT + str(index)] = (entity_value, original_text)
+            index = index + 1
+        return detected_number_dict
+
+    def _get_original_text_from_tagged_text(self, number_tag_text):
+        """
+        Return original text value of number tag from number detected map
+        Args:
+            number_tag_text (str): tagged number
+        Returns:
+            (str or None): Original value of tagged number if found else None
+        """
+        original = number_tag_text
+        for number_tag in self.number_detected_map:
+            original = original.replace(number_tag, self.number_detected_map[number_tag][1])
+        if original == number_tag_text:
+            return None
+        return original
+
+    def detect_number_range(self, text):
+        """
+        Detect number-range from number range text. Run through list of detectors defined in detector_preferences in
+        the preferences.
+        Args:
+            text(str): text string
+        Returns:
+            number_list (list): list containing detected numeric text
+            original_list (list): list containing original numeral text
+
+        """
+        self.text = text
+        self.processed_text = text
+        self.tagged_text = text
+        self.number_detected_map = self._get_number_tag_dict()
+
+        number_list, original_list = None, None
+        for detector in self.detector_preferences:
+            self.number_tagged_processed_text = self._tag_number_in_text(self.processed_text)
```

Comment:
We have to update this, since if we do it once,  if the first detector detect something it will update processed text, which will not get reflected in  number_tagged_processed_text